### PR TITLE
Create PerSegmentBaseModel, PerSegmentPredictionIntervalModel, PerSegmentModel

### LINKED ISSUE #506: Create `PerSegmentBaseModel`, `PerSegmentPredictionIntervalModel`, `PerSegmentModel`
### 🚀 Feature Request

Instead of one `PerSegmentModel` create three models

### Motivation

It will make abstractions easier to comprehend.

### Proposal

Create `PerSegmentBaseModel`
- it should inherit from `FitAbstractModel`
- It should implement method `fit` (actually it is already implemented in current PerSegmentModel)
- It should implement method `init` (already implemented in current PerSegmentModel)
- it should implement method `_build_models` (already implemented in current PerSegmentModel)
- it should implement method `get_models` that returns dict of all segment-models pairs
-  it should implement method `_forecast_segment` (already implemented in current Model), except it should expect additional parameters: `def _forecast_segment(model, segment: Union[str, List[str]], ts: TSDataset, *args, **kwargs) -> pd.DataFrame` and pass them to https://github.com/tinkoff-ai/etna/blob/1a10b256956352dd576ae8972771534c9ad47ed8/etna/models/base.py#L79

Create `PerSegmentPredictionIntervalModel`
- it should inherit from `PerSegmentBaseModel` and `ForecastPredictIntervalAbstractModel`
- It should implement method `forecast`, that passes prediction_interval, quantiles to model's predict method

Create `PerSegmentModel`
- it should inherit from `PerSegmentBaseModel` and `ForecastAbstractModel`
- It should implement method `forecast` (actually it is already implemented in current PerSegmentModel)

[Create](https://pypi.org/project/typing-extensions/) `FitPredictProtocol` as all these classes expect base_model to implement it.
- fit `def fit(self, df: pd.DataFrame, *args, **kwargs)`
- predict `def predict(self, df: pd.DataFrame, *args, **kwargs) -> pd.DataFrame`

Use `FitPredictProtocol` for typing in `PerSegmentBaseModel`: `model: FitPredictProtocol`
DO NOT use FitPredictProtocol for inheritance, FitPredictProtocol should be used only for type checks.

Also change the line
https://github.com/tinkoff-ai/etna/blob/1a10b256956352dd576ae8972771534c9ad47ed8/etna/pipeline/pipeline.py#L156
As it should check isinstance, not signature.

### Test cases

_No response_

### Alternatives

_No response_

### Additional context

Blocked by #505 

Also future developers should note that FitPredictProtocol is for inner library use only. 
User guaranteed only fit-forecast interface stated by #505 and one should not worry about inner implementation of `PerSegmentBaseModel`, `PerSegmentPredictionIntervalModel`, `PerSegmentModel` and model adapters.

### Checklist

- [x] I discussed this issue with ETNA Team

### PR DESCRIPTION
**Pull request by alex-hse-repository**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Closing issues

closes #506

### CODE REVIEW

File: etna/models/base.py [hunk 1]
```diff
@@ -230,3 +166,181 @@ def forecast(
...
+            Dataset with predictions
+        """
+        result_list = list()
+        for segment, model in self.get_model().items():
+            segment_predict = self._forecast_segment(
+                model=model, segment=segment, ts=ts, prediction_interval=prediction_interval, quantiles=quantiles
+            )
+            result_list.append(segment_predict)
+
+        # need real case to test
```
Comment (iKintosh) [inline etna/models/base.py:336]: Is it still relevant?

File: etna/models/sarimax.py [hunk 2]
```diff
@@ -534,7 +534,7 @@ def forecast(
 
         result_list = list()
         for segment in self._segments:
-            model = self._models[segment]
+            model = self._models[segment]  # type: ignore
 
             segment_predict = self._forecast_one_segment(model, segment, ts, prediction_interval, quantiles)
```
Comment (iKintosh) [inline etna/models/sarimax.py:539]: Does it mean, that after this PR SARIMAX no longer support prediction intervals until #507 is done?

**Review by iKintosh — COMMENTED** [review]
Have some question considering Prophet and SARIMAX

File: etna/models/base.py (hunk 1 above)
Comment (alex-hse-repository) [reply etna/models/base.py:336]: No ideas)

File: etna/models/sarimax.py (hunk 2 above)
Comment (alex-hse-repository) [reply etna/models/sarimax.py:539]: No it does't, Prophet and Sarimax have their own fit/forrecast

File: etna/models/base.py (hunk 1 above)
Comment (iKintosh) [reply etna/models/base.py:336]: Let's remove it then

**Review by iKintosh — COMMENTED** [review]
Looks good!
Let's remove deprecated comments and let's merge.

**Review by iKintosh — APPROVED** [review]

**Pull request merged**